### DISCUSSION TITLE
Type inference

### DISCUSSION THREAD

**Discussion by rprunskas** [LemonAppDev/konsist#579] [Ask a Question] [answered]
### Wanted end result.

When code is with type inference, for example:

`private val _documentState = MutableLiveData<DocumentState>()`

Konsist doesn't detect property type, it is null. Should it be like that?

### What you tried so far.

I had to use such filter to get MutableLiveData property. Is there any better way? 

```
  @Test
    fun `viewModel MutableLiveData is private and name starts with _`() {
        Konsist.scopeFromProject()
            .classes()
            .withParentOf(ViewModel::class)
            .properties(includeNested = false, includeLocal = false)
            .filter {
                it.text.contains("MutableLiveData<[^>]+>".toRegex())
            }
            .assert {
                it.hasPrivateModifier && it.hasNameStartingWith("_")
            }
    }
```

**Comment by igorwojda** [comment, marked as answer]
This is expected behavior. See
https://docs.konsist.lemonappdev.com/features/compiler-inference

In general, your test looks good. I think that `includeNested = false, includeLocal = false` arguments are redundant.

BTW I will add to the backlog ticket to add `.withTextContaining` methods (`.withTextContaining("MutableLiveData<[^>]+>".toRegex())`)

**Comment by rprunskas** [comment]
Oh there is a page about it. How can I missed that :) Thanks

**Reply by igorwojda** [reply]
I have updated the docs after your post ;-)
